### PR TITLE
Download books from AA API if userkey is set

### PR DESCRIPTION
**Pull request by FlappyTaps**
This PR uses the Anna's Archive API to get a fast download URL to download the book directly. Bypasses parsing HTML with the Cloudflare proxy.

### CODE REVIEW

**Comment by Aymendje** [comment]
Thank you for the PR ! 
I was just looking into it and had an initial draft too :
https://github.com/calibrain/calibre-web-automated-book-downloader/pull/16/files
I don't mind dropping mine and merging yours, but could you change a bit the logic in book_manager.py to reuse basically what I have ?

This way, if for whatever reason we fail to get the book from the AA_API, we can still retry using the "old" method

**Comment by calibrain** [comment]
Thank you for the PR !
I was just looking into it and had an initial draft too :
https://github.com/calibrain/calibre-web-automated-book-downloader/pull/16/files
I don't mind dropping mine and merging yours, but could you change a bit the logic in book_manager.py to reuse basically what I have ?

This way, if for whatever reason we fail to get the book from the AA_API, we can still retry using the "old" method

File: Dockerfile [hunk 1]
```diff
@@ -17,6 +17,7 @@ ENV CLOUDFLARE_PROXY_URL=http://localhost:8000
 ENV INGEST_DIR=/cwa-book-ingest
 ENV STATUS_TIMEOUT=3600
 ENV PYTHONPATH=/app
+ENV AA_DONATOR_KEY=None
```
Comment (calibrain) [inline Dockerfile]: Let's just leave this one unset

**Review by calibrain — CHANGES_REQUESTED** [review]
Awesome ! One small change and then ill merge !

**Comment by FlappyTaps** [comment]
Done :)

**Comment by calibrain** [comment]
Oh, can you please completely remove it from the dockerfile ?

By default, if it exist (even if empty) it will be processed in python as `''` (empty string) and not as `None` so it will always try it, even if its empty

**Comment by calibrain** [comment]
Awesome, thank you for the work !

**Pull request merged**